pkg/internal/polymorphichelpers/rollout_status.go

```diff
@@ -273,9 +273,13 @@ func (s *CloneSetStatusViewer) Status(c kubernetes.Interface, obj runtime.Unstru
 	if cs.Spec.UpdateStrategy.Type == kruiseappsv1alpha1.InPlaceOnlyCloneSetUpdateStrategyType ||
 		cs.Spec.UpdateStrategy.Type == kruiseappsv1alpha1.InPlaceIfPossibleCloneSetUpdateStrategyType {
 		if cs.Spec.Replicas != nil && cs.Spec.UpdateStrategy.Partition != nil {
-			return fmt.Sprintf("Waiting for partitioned roll out to finish: %d out of %d new pods have been updated...\n",
-				cs.Status.UpdatedReplicas, *cs.Spec.Replicas-cs.Spec.UpdateStrategy.Partition.IntVal), false, nil
+			if cs.Status.UpdatedReplicas < (*cs.Spec.Replicas - cs.Spec.UpdateStrategy.Partition.IntVal) {
```

Comment:
Partition can be percentage, so we can not just use the `IntVal` here. You can see how latest CloneSet calculates the real partition number: https://github.com/openkruise/kruise/blob/3153179fdd236a1942d7f292a08f44774da58e5c/pkg/util/tools.go#L155